Sources/Datasets/datasets/image/MNIST.swift

```diff
@@ -38,9 +40,53 @@ public struct MNISTIterator: S5TFDatasetIterator {
 
 public struct MNIST: S5TFDataset {
     typealias _Iterator = MNISTIterator // swiftlint:disable:this type_name
-    internal let data = [[1, 2], [3, 4], [5, 6]]
+
+    internal var data = [[1, 2], [3, 4], [5, 6]]
+
     public func batched(_ batchSize: Int) -> MNISTIterator {
         return MNISTIterator(dataset: self)
     }
+
+    public var split: S5TFSplit
+    init() { self.split = .undefined }
+    private init(split: S5TFSplit) {
+        self.split = split
+        let downloader = Downloader()
+        func getTrain() {
+            downloader.download(fileAt: URL(string: "https://storage.googleapis.com/cvdf-datasets/mnist/train-images-idx3-ubyte.gz")!) { url, error in }
+            downloader.download(fileAt: URL(string: "https://storage.googleapis.com/cvdf-datasets/mnist/train-labels-idx1-ubyte.gz")!) { url, error in }
+        }
+        func getTest() {
+            downloader.download(fileAt: URL(string: "https://storage.googleapis.com/cvdf-datasets/mnist/t10k-images-idx3-ubyte.gz")!) { url, error in }
+            downloader.download(fileAt: URL(string: "https://storage.googleapis.com/cvdf-datasets/mnist/t10k-labels-idx1-ubyte.gz")!) { url, error in }
+        }
+        switch split {
+            case .train:
+                getTrain()
+            case .test:
+                getTest()
+            case .all:
+                getTrain()
+                getTest()
+            default:
+                break            
+        }
+    }
+    var train : MNIST {
+        guard self.split == .undefined else { fatalError("this property can only be accessed from an undefined split") }
+        return MNIST(split: .train)
+    }
+    var validation : MNIST {
+        fatalError("the split does not exist for this dataset")
+    }
+    var test : MNIST {
+        guard self.split == .undefined else { fatalError("this property can only be accessed from an undefined split") }
+        return MNIST(split: .test)
+    }
+    var all : MNIST {
+        guard self.split == .undefined else { fatalError("this property can only be accessed from an undefined split") }
+        return MNIST(split: .all)
+    }
+
```

Comment:
Consider placing variables at the top. They can be grouped by adding a single blank line under a coherent group.